lang/java/tools/src/main/java/org/apache/avro/tool/SpecificCompilerTool.java

```diff
@@ -164,7 +181,7 @@ public String getShortDescription() {
       System.err.println("No input files found.");
     }
 
-    return fileSet.toArray((new File[fileSet.size()]));
+    return fileSet.toArray(new File[0]);
```

Comment:
Why did you change this? Since we know the size of the result array, the original code (allocating one before toArray, and passing it) is still fine isn't it?

Reply:
[Here is an excellent article](https://shipilev.net/blog/2016/arrays-wisdom-ancients/) that explains why nowadays it's better (despite counter intuitive) to use `toArray(new File[0])` in favor of `toArray(new File[fileSet.size()])`. The old code was not wrong, but I wanted to remove those redundant parentheses and switched to `new File[0]` because I was already editing that line. ;-)

Reply:
@nandorKollar If you prefer `toArray(new File[fileSet.size()])`, I can revert it of course.